.github/check-c-formatting.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env sh
+# I wanted to get the merge base using {{ github.base_ref }}, however this is only
+# available for the event that opens the PR or edits it, not on pushes to the branch.
+# Comparing to main should be an OK alternative, since it will - at worst - do more
+# autoformatting than it otherwise would.
+merge_base=main
```

Comment:
if you add a
`[ -z "$1" ] && merge_base=main || merge_base="$1"` then you could do that when available while setting main when not.